packages/bloc/lib/src/bloc.dart

```diff
@@ -41,14 +43,20 @@ abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
     void Function() onDone,
     bool cancelOnError,
   }) {
-    return _stateSubject.listen(
+    return _createStateStream().listen(
       onData,
       onError: onError,
       onDone: onDone,
       cancelOnError: cancelOnError,
     );
   }
 
+  // Creates a Stream that emits the current state and any future states
+  Stream<State> _createStateStream() async* {
```

Comment:
One thing that came up for me: How important is it that blocs emit the latest event when you first listen? Is it even necessary?

I know that'd be a bigger breaking change to discuss, but seems like most blocs are hooked up to `StreamBuilder` widgets. Since you already have the `initialState` to pass to the `StreamBuilder` for the initial build, is it really necessary for this class to emit the latest value to all listeners (triggering 2 builds for each BlocBuilder widget?)

Reply:
Haha I was thinking about that yesterday as well. I think keeping the behavior as is for now makes sense. I need to think through what the implications of removing that are because actually BlocBuilder and BlocListener already use skip(1) to avoid the 2 builds issue.